components/function-controller/internal/controllers/serverless/function_reconcile.go

```diff
@@ -169,19 +169,23 @@ func (r *FunctionReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error
 
 	gitOptions, err := r.readGITOptions(ctx, instance)
 	if err != nil {
-		return r.updateStatusWithoutRepository(ctx, ctrl.Result{}, instance, serverlessv1alpha1.Condition{
+		if updateErr := r.updateStatusWithoutRepository(ctx, instance, serverlessv1alpha1.Condition{
 			Type:               serverlessv1alpha1.ConditionConfigurationReady,
 			Status:             corev1.ConditionFalse,
 			LastTransitionTime: metav1.Now(),
 			Reason:             serverlessv1alpha1.ConditionReasonSourceUpdateFailed,
 			Message:            fmt.Sprintf("Reading git options failed: %v", err),
-		})
+		}); updateErr != nil {
+			return ctrl.Result{}, errors.Wrap(updateErr, "while updating status")
```

Comment:
Put the error log here as it will be forgotten.

Reply:
Done as requested. But, FWIW, this is inconsistent and not a good practice